// angular-deezer-api/angular-deezer-api.umd.js
(function (global, factory) {
	typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('@angular/common'), require('@angular/http'), require('rxjs/add/operator/toPromise')) :
	typeof define === 'function' && define.amd ? define(['exports', '@angular/core', '@angular/common', '@angular/http', 'rxjs/add/operator/toPromise'], factory) :
	(factory((global['angular-deezer-api'] = {}),global._angular_core,global._angular_common,global._angular_http));
}(this, (function (exports,_angular_core,_angular_common,_angular_http) { 'use strict';

var DeezerApiService = (function () {
    /**
     * @param {?} http
     * @param {?} _jsonp
     */
    function DeezerApiService(http, _jsonp) {
        this.http = http;
        this._jsonp = _jsonp;
        this.searchUrl = 'https://api.deezer.com/search?output=jsonp&callback=JSONP_CALLBACK&q=';
    }
    /**
     * @param {?} value
     * @return {?}
     */
    DeezerApiService.prototype.search = function (value) {
        var /** @type {?} */ url = this.searchUrl + value;
        return this._jsonp.get(url)
            .toPromise()
            .then(function (response) {
            return response.json();
        })
            .catch(this.handleError);
    };
    /**
     * @param {?} error
     * @return {?}
     */
    DeezerApiService.prototype.handleError = function (error) {
        return Promise.reject(error.message || error);
    };
    return DeezerApiService;
}());
DeezerApiService.decorators = [
    { type: _angular_core.Injectable },
];
/**
 * @nocollapse
 */
DeezerApiService.ctorParameters = function () { return [
    { type: _angular_http.Http, },
    { type: _angular_http.Jsonp, },
]; };

var DeezerApiModule = (function () {
    function DeezerApiModule() {
    }
    /**
     * @return {?}
     */
    DeezerApiModule.forRoot = function () {
        return {
            ngModule: DeezerApiModule,
            providers: [DeezerApiService]
        };
    };
    return DeezerApiModule;
}());
DeezerApiModule.decorators = [
    { type: _angular_core.NgModule, args: [{
                imports: [
                    _angular_common.CommonModule,
                    _angular_http.HttpModule,
                    _angular_http.JsonpModule
                ],
                declarations: [],
                exports: []
            },] },
];
/**
 * @nocollapse
 */
DeezerApiModule.ctorParameters = function () { return []; };

exports.DeezerApiModule = DeezerApiModule;
exports.DeezerApiService = DeezerApiService;

Object.defineProperty(exports, '__esModule', { value: true });

})));
